sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/EliminateResolvedHint.scala

```diff
@@ -37,7 +41,7 @@ object EliminateResolvedHint extends Rule[LogicalPlan] {
     }
     pulledUp.transformUp {
       case h: ResolvedHint =>
-        handleInvalidHintInfo(h.hints)
+        hintErrorHandler.handleJoinNotFoundForJoinHint(h.hints)
```

Comment:
This seems not join specific, maybe `handleInvalidHintInfo` is a better name.

Reply:
No. The original name is not accurate. This rule basically extract hints nodes from child relations of a join and pull them up to the Join node. If we come up with other "invalid hints" later, we can have other error handler callbacks specific to those invalid hints.

Reply:
This rule is called `EliminateResolvedHint`, is it possible that we add other hints and also report hint target not found issue here?

Reply:
The hope is we'll be able to recognize the specific hint error in this rule if anything else. Say if a XYZ-hint has to be associated with XYZ-node, we should another error assigned to it.